File: t/VB/GrayScaleDemo/UI/Sigma.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************

Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools.ImageProcessing.Core
Imports System

Partial Public Class SigmaDialog
   Inherits Form
   Private Shared _dimension As Integer
   Private Shared _outline As Boolean
   Private Shared _threshold As Single
   Private Shared _sigma As Integer

   Public ReadOnly Property Dimension() As Integer
      Get
         Return _dimension
      End Get
   End Property

   Public ReadOnly Property Sigma() As Integer
      Get
         Return SigmaDialog._sigma
      End Get
   End Property

   Public ReadOnly Property Threshold() As Single
      Get
         Return SigmaDialog._threshold
      End Get
   End Property

   Public ReadOnly Property Outline() As Boolean
      Get
         Return SigmaDialog._outline
      End Get
   End Property

   Public Sub New()
      InitializeComponent()
   End Sub

   Private Sub Sigma_Load(sender As Object, e As EventArgs) Handles Me.Load
      Dim cmd As New SigmaCommand()
      _dimension = cmd.Dimension
      _outline = cmd.Outline
      _threshold = cmd.Threshold
      _sigma = cmd.Sigma

      _numDimension.Value = _dimension
      _numThreshold.Value = CDec(_threshold)
      _numSigma.Value = _sigma
      _cbOutline.Checked = _outline
   End Sub

   Private Sub _btnOk_Click(sender As Object, e As EventArgs) Handles _btnOk.Click
      _dimension = CInt(_numDimension.Value)
      _threshold = CSng(_numThreshold.Value)
      _sigma = CInt(_numSigma.Value)
      _outline = _cbOutline.Checked
   End Sub
End Class

File: t/CS/GrayScaleDemo/UI/Sigma.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools.ImageProcessing.Core;

namespace GrayScaleDemo
{
    public partial class SigmaDialog : Form
    {
        private static int _dimension;
        private static bool _outline;
        private static float _threshold;
        private static int _sigma;

        public int Dimension
        {
            get { return _dimension; }
        }

        public int Sigma
        {
            get { return SigmaDialog._sigma; }
        }

        public float Threshold
        {
            get { return SigmaDialog._threshold; }
        }

        public bool Outline
        {
            get { return SigmaDialog._outline; }
        }
       
        public SigmaDialog()
        {
            InitializeComponent();
        }

        private void Sigma_Load(object sender, EventArgs e)
        {
             SigmaCommand cmd = new SigmaCommand();
             _dimension = cmd.Dimension;
             _outline = cmd.Outline;
             _threshold = cmd.Threshold;
             _sigma = cmd.Sigma;

            _numDimension.Value      = _dimension;
            _numThreshold.Value      = (decimal)_threshold;
            _numSigma.Value          = _sigma;
            _cbOutline.Checked       = _outline;
        }

        private void _btnOk_Click(object sender, EventArgs e)
        {
            _dimension = (int)_numDimension.Value;
            _threshold=(float)_numThreshold.Value  ;
            _sigma = (int)_numSigma.Value;
            _outline = _cbOutline.Checked;
        }
    }
}
